homework.3/homework.3.playground/Contents.swift

```diff
@@ -0,0 +1,116 @@
+// homework.3
+
+import Darwin
+
+addNumbers(10, 12)
+addNumbers(99.8, 100.2)
+
+subNumbers(100, 50)
+subNumbers(200.5, 100)
+
+divNumbers(20, 4)
+divNumbers(12.8, 2.0)
+
+mulNumbers(2, 8)
+mulNumbers(20.0, 5.1)
+
+sumDigits(1234)
+
+compStrings("Hello", "!")
+
+exponentiation(-2, 5)
+
+print("Factorial = \(factorial(-1))")
+
+// Addition Functions
+
+func addNumbers(_ firstNumber: Int, _ secondNumber: Int) {
+    print("intNumberAddition = \(firstNumber + secondNumber)\n")
+}
+
+func addNumbers(_ firstNumber: Double, _ secondNumber: Double) {
+    print("doubleNumberAddition = \(firstNumber + secondNumber)\n")
+}
+
+// Subtraction Functions
+
+func subNumbers(_ firstNumber: Int, _ secondNumber: Int) {
+    print("intNumberSubtraction = \(firstNumber - secondNumber)\n")
+}
+
+func subNumbers(_ firstNumber: Double, _ secondNumber: Double) {
+    print("doubleNumberSubtraction = \(firstNumber - secondNumber)\n")
+}
+
+// Multiplication Functions
+
+func mulNumbers(_ firstNumber: Int, _ secondNumber: Int) {
+    print("intNumberMultiplication = \(firstNumber * secondNumber)\n")
+}
+
+func mulNumbers(_ firstNumber: Double, _ secondNumber: Double) {
+    print("doubleNumberMultiplication = \(firstNumber * secondNumber)\n")
+}
+
+// Division Functions
+
+func divNumbers(_ firstNumber: Int, _ secondNumber: Int) {
+    print("intNumberDivision = \(firstNumber / secondNumber)\n")
+}
+
+func divNumbers(_ firstNumber: Double, _ secondNumber: Double) {
+    print("doubleNumberDivision = \(firstNumber / secondNumber)\n")
+}
```

Comment:
It's also better to use full words not shortened in names. So `subtractNumbers`, `multiplyNumbers`, `divideNumbers`, not `subNumbers`, `mulNumbers`, `divNumbers`